pkgs/by-name/ya/yazi/package.nix

```diff
@@ -4,25 +4,13 @@
   runCommand,
   makeWrapper,
 
-  extraPackages ? [ ],
-  optionalDeps ? [
-    jq
-    poppler_utils
-    _7zz
-    ffmpegthumbnailer
-    fd
-    ripgrep
-    fzf
-    zoxide
-    imagemagick
-    chafa
-  ],
+  runtimeDeps ? (ps: ps),
```

Comment:
Why is this a function, and how is this making it easier to override?
If I want to remove the optional deps, I'd have to write `yazi.override { runtimeDeps = _: []; }`? That just seems confusing/unintuitive.

Reply:
I think it makes it easier to substitute or add a dep on top of the default deps, which as far as I can tell wasn't possible before?

Reply:
I guess it works, but it's using a pattern I've never personally observed in nixpkgs. If nixpkgs used the module system this would be a lot easier to do...